modules/haberdasher-labs_public-lock-v14-eligibility-hatter_v0.1.2.json

```diff
@@ -0,0 +1,625 @@
+{
+  "id": "haberdasher-labs_public-lock-v14-eligibility-hatter",
+  "version": "v0.1.2",
+  "name": "Subscription Eligibility (Unlock Protocol V14)",
+  "details": [
+    "This module adds a subscription requirement to a hat. Wearers must make a recurring payment to keep the hat.",
+    "This module is powered by Unlock Protocol. Deploying this module will also create a new lock contract (Public Lock version 14), and will automatically mint a hat to the recipient of a newly purchased key. The wearer will remain eligible as long as their subscription is active.",
+    "To work properly, this module must be set as the eligibility module for the target hat and wear one of the target hat's admin hats."
+  ],
+  "links": [
+    {
+      "label": "GitHub",
+      "link": "https://github.com/Hats-Protocol/unlock-eligibility"
+    }
+  ],
+  "parameters": [
+    {
+      "label": "Referrer",
+      "functionName": "REFERRER",
+      "displayType": "default"
+    },
+    {
+      "label": "Referrer Fee Percentage",
+      "functionName": "referrerFeePercentage",
+      "displayType": "default"
+    },
+    {
+      "label": "Lock Contract",
+      "functionName": "lock",
+      "displayType": "default"
+    },
+    {
+      "label": "Price",
+      "functionName": "keyPurchasePrice",
+      "displayType": "amountWithDecimals"
+    },
+    {
+      "label": "Token",
+      "functionName": "keyPurchaseToken",
+      "displayType": "token"
+    },
+    {
+      "label": "Renewal Period",
+      "functionName": "expirationDuration",
+      "displayType": "seconds"
+    },
+    {
+      "label": "Max Purchasers",
+      "functionName": "maxNumberOfKeys",
+      "displayType": "default"
+    }
+  ],
+  "type": {
+    "eligibility": true,
+    "toggle": false,
+    "hatter": true
+  },
+  "tags": [],
+  "implementationAddress": "0x4c7803041851f7a17Fc6b5Ff5c911FC748160637",
+  "deployments": [
+    {
+      "chainId": "11155111",
+      "block": "6699345"
+    },
+    {
+      "chainId": "10",
+      "block": "125422530"
+    },
+    {
+      "chainId": "137",
+      "block": "61873232"
+    },
+    {
+      "chainId": "42161",
+      "block": "253931725"
+    },
+    {
+      "chainId": "100",
+      "block": "36021503"
+    },
+    {
+      "chainId": "8453",
+      "block": "19827469"
+    },
+    {
+      "chainId": "42220",
+      "block": "27754441"
+    }
+  ],
+  "creationArgs": {
+    "useHatId": true,
+    "immutable": [],
+    "mutable": [
+      {
+        "name": "Subscription Period",
+        "description": "How often a payment must be made to keep the hat",
+        "type": "uint256",
+        "example": "2592000",
+        "displayType": "seconds"
+      },
+      {
+        "name": "Payment Token",
+        "description": "The token used to make payments",
+        "type": "address",
+        "example": "0xb16f35c0ae2912430dac15764477e179d9b9ebea",
+        "displayType": "token"
+      },
```

Comment:
This can also be Eth if the address is set to zero, should we add to the description?

Reply:
> a lock whose currency is ETH, which works but is not ideal beecause it does not allow for automated renewals.

per Julien ETH doesn't support automated renewals. we likely won't support ETH in the app, fwiw